openmc/plotter.py

```diff
@@ -56,18 +55,61 @@
 ELEMENT_NAMES = list(openmc.data.ELEMENT_SYMBOL.values())[1:]
 
 
-def plot_xs(this, types, divisor_types=None, temperature=294., axis=None,
+def _get_legend_label(this, type):
+    """Gets a label for the element or nuclide or material and reaction plotted"""
+    if isinstance(this, str):
+        return f'{this} {type}'
+    elif this.name is '':
+        return f'material name not set {type}'
+    else:
+        return f'{this.name} {type}'
+
+
+def _get_yaxis_label(this_and_types, divisor_types):
+    """Gets a y axis label for the type of data plotted"""
+
+    if all(isinstance(item, str) for item in this_and_types.keys()):
+        stem = 'Microscopic'
+        if divisor_types:
+            mid, units = 'Data', ''
+        else:
+            mid, units = 'Cross Section', '[b]'
+    elif all(isinstance(item, openmc.Material) for item in this_and_types.keys()):
+        stem = 'Macroscopic'
+        if divisor_types:
+            mid, units = 'Data', ''
+        else:
+            mid, units = 'Cross Section', '[1/cm]'
+    else:
+        msg = "Mixture of openmc.Material and elements/nuclides. Invalid type for plotting"
+        raise TypeError(msg)
+
+    return f'{stem} {mid} {units}'
+
+
+def _get_title(this_and_types):
+    """Gets a title for the type of data plotted"""
+    if len(this_and_types) == 1:
+        this = list(this_and_types.keys())[0]
+        name = this.name if isinstance(this, openmc.Material) else this
+        return f'Cross Section Plot For {name}'
+    else:
+        return 'Cross Section Plot'
+
+
+def plot_xs(this_and_types, divisor_types=None, temperature=294., axis=None,
             sab_name=None, ce_cross_sections=None, mg_cross_sections=None,
             enrichment=None, plot_CE=True, orders=None, divisor_orders=None,
             **kwargs):
     """Creates a figure of continuous-energy cross sections for this item.
 
     Parameters
     ----------
-    this : str or openmc.Material
-        Object to source data from. Nuclides and elements can be input as a str
-    types : Iterable of values of PLOT_TYPES
-        The type of cross sections to include in the plot.
+
+    this_and_types : dict where keys are str or openmc.Material, values are Iterable of PLOT_TYPES
```

Comment:
As long as we're changing the function signature (which I'm ok with -- I don't think this one gets a ton of usage at present), can we come up with a better name than `this_and_types`?

My only hesitance about the change (other than naming issues) is that this makes the simple case (single item/reaction) a bit more complex. But, the added functionality is probably worth it. :man_shrugging:

For documentation purposes, what comes after `:` here needs to be a proper type
```suggestion
    this_and_types : dict
```